cmd/demo/demo_test.go

```diff
@@ -24,72 +21,61 @@ var (
 	timeout = time.Second * 30
 )
 
+type Cmd struct {
+	*exec.Cmd
+	stdin io.WriteCloser
+}
+
+func nodeCmd(name string) (*Cmd, error) {
+	args := []string{
+		"run", "../../main.go",
+		"demo",
+		"--config", fmt.Sprintf("../../%v.yaml", name),
+		"--network", "../../network.yaml",
+		"--log-level", "trace",
+		"--log-file", fmt.Sprintf("%v.log", name),
+		"--stdio",
+	}
+	cmd := exec.Command("go", args...)
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return nil, err
+	}
+	return &Cmd{cmd, stdin}, nil
+}
+
 func TestNodes(t *testing.T) {
-	alice, _, err := expect.Spawn("go run ../../main.go demo --config ../../alice.yaml --network ../../network.yaml --log-level trace --test-api true --log-file alice.log", -1)
+	// Start Alice
+	alice, err := nodeCmd("alice")
 	require.NoError(t, err)
-	defer alice.Close()
-	time.Sleep(time.Second * 2)
+	require.NoError(t, alice.Start())
+	defer alice.Process.Kill()
+	time.Sleep(5 * time.Second)
```

Comment:
If we remove the blocktime, we can shorten the timeouts.

Reply:
ok, but isn't it useful to test with blocktime? at the moment the test is designed to work with the same ganache setup as described in the README.

Reply:
Ok then lets leave it like this, makes indeed more sense to test with block time.

Reply:
I just noticed that we wait 5s (one blocktime) here, but the contract deployment of Alice (Adjudicator+AH) takes two blocks.  
So we actually have to wait 10s here, which then also fixes the test for me.

Reply:
that makes sense and also explains why i didn't get the error. i was usually starting the test after i already manually test something else, so the contracts were already deployed... will fix this.